Source/Core/Common/Network.h

```diff
@@ -99,8 +115,124 @@ struct UDPHeader
 };
 static_assert(sizeof(UDPHeader) == UDPHeader::SIZE);
 
+#pragma pack(push, 1)
+struct ARPHeader
+{
+  ARPHeader();
+  ARPHeader(u32 from_ip, MACAddress from_mac, u32 to_ip, MACAddress to_mac);
+  u16 Size() const;
+
+  static constexpr std::size_t SIZE = 28;
+
+  u16 hardware_type = 0;
+  u16 protocol_type = 0;
+  u8 hardware_size = 0;
+  u8 protocol_size = 0;
+  u16 opcode = 0;
+  MACAddress sender_address{};
+  u32 sender_ip = 0;
+  MACAddress targer_address{};
+  u32 target_ip = 0;
+};
+static_assert(sizeof(ARPHeader) == ARPHeader::SIZE);
+#pragma pack(pop)
+
+struct DHCPBody
+{
+  DHCPBody();
+  DHCPBody(u32 transaction, MACAddress client_address, u32 new_ip, u32 serv_ip);
+  DHCPBody(std::vector<u8> data);
+  bool AddDHCPOption(u8 size, u8 fnc, const std::vector<u8>& params);
+  static constexpr std::size_t SIZE = 540;
+  u8 message_type = 0;
+  u8 hardware_type = 0;
+  u8 hardware_addr = 0;
+  u8 hops = 0;
+  u32 transaction_id = 0;
+  u16 secondes = 0;
+  u16 boot_flag = 0;
+  u32 client_ip = 0;
+  u32 your_ip = 0;
+  u32 server_ip = 0;
+  u32 relay_ip = 0;
+  MACAddress client_mac{};
+  unsigned char padding[10]{};
+  unsigned char hostname[0x40]{};
+  unsigned char boot_file[0x80]{};
+  u32 magic_cookie = 0x63538263;
+  u8 options[300]{};
```

Comment:
The option size should be 312, including the magic_cookie. I chose to use `u8[4]` as it doesn't depend on endianness, using `u32` is slightly confusing.